### DISCUSSION TITLE
How to leverage `tolerance` in sel()

### DISCUSSION THREAD

**Discussion by marstonsward** [pydata/xarray#7544] [Q&A] [answered]
I am using the `sel()` to locate the data in an 3D data array. 
But I would like to know if I can use the `tolerance` argument to be used in 3D.
Here's my use case:

`gtvec = gtemp.sel(lat=lat, lon=lon, method="nearest", tolerance=0.25, drop=False)' 

Here `tolerance=0.25` is used for both lat and lon, But I would like to search in the `lev=0.5` dimension as well in this fashion:

`gtvec = gtemp.sel(lev=lev, lat=lat, lon=lon, method="nearest", tolerance=0.25, drop=False)`
But can `tolerance` take a `list` or `dict` for each dimension? It is not clear in the docs.

**Comment by benbovy** [comment, marked as answer]
Hi @Marston, there's no way currently of setting a tolerance value per coordinate/index, apart from chaining multiple  `.sel()` calls for `lat`/`lon` and then for `lev`. See also #3223.

There is a related discussion in #7099 about how to pass arbitrary options to `.sel()`, which is a bit more general than your use case but that would make it possible.

**Reply by marstonsward** [reply]
@benbovy 

Thank you. I understand. I'll chain the statements.
